File: Wirtualny swiat-java/Animals/Antilope.java
package Animals;

import World.Organism;
import World.Animal;
import World.GameWorld;
import World.Point;

import java.util.Random;

public class Antilope extends Animal {
    private final Random random = new Random();

    public Antilope(int x, int y, GameWorld world) {

        super(4, 4, x, y, world, 'A');
    }

    @Override
    public void draw() {

        System.out.print(symbol);
    }

    @Override
    public void action() {

        super.action();
    }

    @Override
    public Point newPosition() {
        int oldX = getPositionX();
        int oldY = getPositionY();
        int newX, newY;

        do {
            newX = oldX + (random.nextInt(3) - 1) * 2;  // -2, 0, 2
            newY = oldY + (random.nextInt(3) - 1) * 2;
        } while (!gameWorld.isOkay(newX, newY) || (newX == oldX && newY == oldY));

        return new Point(newX, newY);
    }

    @Override
    public boolean isBlockAttack(Organism attacker) {
        int randNum=random.nextInt(2);
        if(randNum==0)
            return true;
        else
            return false;
    }
}

File: Wirtualny Swiat C++/animals/Antilope.cpp
//
// Created by Dorota on 17.04.2025.
//

#include "Antilope.h"
#include "../GameWorld.h"
#include <iostream>
using namespace std;

Antilope::Antilope(int x, int y, GameWorld *world): Animal(4, 4, x, y, world, 'A') {

}

void Antilope::draw() {
    cout<<symbol;
}

void Antilope::action() {
    Animal::action(); //dziedziczenie ruchu
}

Point Antilope::newPostion() {
    int newX=0, newY=0;
    int oldPosX = GetPositionX();
    int oldPosY = GetPositionY();

    do {
        newX = oldPosX + (rand()%3-1)*2; //-1 1
        newY = oldPosY + (rand()%3-1)*2;
    }while (gameWorld->isOkay(newX, newY)==false || (newX==oldPosX && newY==oldPosY));
    return {newX, newY};
}

bool Antilope::isBlockAttack(Organism *attacker) {
    int randNum = rand()%2; //50 procent szans
    if (randNum==0) {
        return true;
    }
    return false;
}
